av/container/output.pyx

```diff
@@ -64,14 +65,12 @@ cdef class OutputContainer(Container):
 
         if codec_name is not None:
             codec_obj = codec_name if isinstance(codec_name, Codec) else Codec(codec_name, 'w')
-            codec = codec_obj.ptr
 
         else:
-            if not template._codec:
-                raise ValueError("template has no codec")
-            if not template._codec_context:
+            if not template.codec_context:
                 raise ValueError("template has no codec context")
-            codec = template._codec
+            codec_obj = template.codec_context
```

Comment:
```suggestion
            codec_obj = template.codec_context.codec
```

Reply:
Ehm. So this basically means we don't have any test coverage for this..